File: csharp/Validators/ArgumentCountValidator.cs
using System;
using System.Collections.Generic;

namespace Validators
{
    public class ArgumentCountValidator: AbstractValidator
    {
        protected ushort Count { get; set; }

        public ArgumentCountValidator(List<string> args, ushort count, string message) : base(args, message)
        {
            Count = count;
        }

        public override bool Validate()
        {
            if (Arguments.Count != Count)
            {
                Console.WriteLine(Message);
                return false;
            }

            return true;
        }
    }
}

File: python/Validators/ArgumentCountValidator.py
from Abstractions.AbstractListValidator import *


class ArgumentCountValidator(AbstractListValidator):
    def __init__(self, validate_this_list, count, message):
        super().__init__(validate_this_list, count, message)

    def validate(self):
        if self._validate_this_list.__len__() != self._count:
            self._print_validation_error()
            return False
        else:
            return True
